packages/contracts/contracts/proofVerifiers/BaseProofVerifier.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IVerifier} from '../interfaces/IVerifier.sol';
+
+contract BaseProofVerifier {
+    IVerifier verifier;
+
+    uint256 public constant SNARK_SCALAR_FIELD =
+        21888242871839275222246405745257275088548364400416034343698204186575808495617;
+
+    struct EpochKeySignals {
+        bool revealNonce;
```

Comment:
Try to put this `bool` near the other small fields. In the EVM each word is 256 bits. When possible the compiler will combine nearby fields into a single work. A `uint8`, `uint48`, `uint160` and `bool` can all be in 1 word.